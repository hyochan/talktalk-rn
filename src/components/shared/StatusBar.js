// @flow
import React from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  Image,
  Text,
  View,
  Platform,
  StatusBar,
} from 'react-native';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import { ratio, colors } from '../../utils/Styles';

type Props = {
  isDarkContent: boolean,
};

type State = {};

function Shared(props: Props, state: State) {
  const statusColor: 'default' | 'light-content' | 'dark-content' = Platform.OS === 'android'
    ? 'default'
    : props.isDarkContent
      ? 'dark-content'
      : 'light-content';
  return (
    <StatusBar
      barStyle={statusColor}
      backgroundColor={colors.darkBlue}
    />
  );
}

Shared.defaultProps = {
  isDarkContent: false,
};
export default Shared;
