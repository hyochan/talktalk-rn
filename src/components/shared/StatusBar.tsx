import React from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  Image,
  Text,
  View,
  Platform,
  StatusBar,
} from 'react-native';
import { ThemeConsumer } from 'styled-components/native';

interface IProps {
  isDarkContent: boolean;
}

interface IState {}

function Shared(props: IProps, state: IState) {
  const statusColor: 'default' | 'light-content' | 'dark-content' = Platform.OS === 'android'
    ? 'default'
    : props.isDarkContent
      ? 'dark-content'
      : 'light-content';
  return (
    <ThemeConsumer>
      {({ colors: { dodgerBlue }}) => (
        <StatusBar
          barStyle={statusColor}
          backgroundColor={dodgerBlue}
        />
      )}      
    </ThemeConsumer>
  );
}

Shared.defaultProps = {
  isDarkContent: false,
};

export default Shared;
