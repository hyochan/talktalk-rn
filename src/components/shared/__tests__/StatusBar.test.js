import 'react-native';
import * as React from 'react';
import StatusBar from '../StatusBar';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';

const component: React.Element<any> = <StatusBar />;

describe('[UserListItem] rendering test', () => {
  it('renders as expected', () => {
    const json = renderer.create(component).toJSON();
    expect(json).toMatchSnapshot();
  });
  it('renders [isDarkContent] as expected', () => {
    const disabledComponent: React.Element<any> = <StatusBar isDarkContent={true}/>;
    const json = renderer.create(disabledComponent).toJSON();
    expect(json).toMatchSnapshot();
  });
});
