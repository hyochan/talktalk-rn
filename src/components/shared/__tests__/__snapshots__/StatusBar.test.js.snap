// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[UserListItem] rendering test renders [isDarkContent] as expected 1`] = `null`;

exports[`[UserListItem] rendering test renders as expected 1`] = `null`;
